doc/sonic-build-system/web_file_server_population_script.md

```diff
@@ -0,0 +1,127 @@
+|-------------	|------------------------------------------------------------------------------------------------------------	|--------------	|---------	|------------------------------------	|
+| source      	| Search path for source tree where to  look for<br> compilation artifacts (version files)                   	| -s, --source 	| String  	| CWD<br>(current working directory) 	|
+| cache       	| Path for temporary storage where to download Web Packages                                                  	| -c, --cache  	| String  	| CWD/tmp                            	|
+| output      	| Output file name to hold the list of packages.<br> The file will be created in the format of version files 	| -o, --output 	| String  	| none                               	|
+| print       	| Print level verbosity                                                                                      	| -p, --print  	| integer 	| 1 (info level)                     	|
+| user        	| User for trusted server authentication                                                                     	| -u, --user   	| String  	| none                               	|
+| key         	| key server authentication                                                                                  	| -k, --key    	| String  	| none                               	|
+| Destination 	| URL for destination web file server                                                                        	| -d, --dest   	| String  	| none                               	|
+
+
+## Script outputs:
+
+The script would print out to STDOUT its current download progress (progress bar) and error messages in case of errors.
+The script can produce a list of packages that were uploaded in a format of version file.
+The script would return 0 in case of success and an errors code (see table below) in case of failure.
+### Error codes:
+
+Code         | Description					        | Comments
+------------ | -------------------------------------| -------------
+ 0		     | Success						        |
+-1		     | Cannot write to cache                |
+-2		     | Cannot access trusted server         |
+-3		     | Cannot authenticate trusted server   |
+
+
+## Build flow integration:
+Script will be integrated into SONiC build environment as part of “make freeze” where invocation of the script is optional. User may choose to call script by defining following environment variables:
+
+```sh
+export SONIC_REPRODUCABLE_BUILD_UPLAOD_ENABLE=1
+export SONIC_REPRODUCABLE_BUILD_UPLAOD_USER=<User for trusted server authentication>
+export SONIC_REPRODUCABLE_BUILD_UPLAOD_KEY=<API key server authentication>
+export SONIC_REPRODUCABLE_BUILD_UPLAOD_DEST=< URL for destination web file server>
+
+```
+## Source tree location:
+ 
+ The script will be located under
+sonic-buildimage/scripts/populate_file_web_server.py
+
+## Q&A
+Who should run this script?

```

Comment:
I believe reproducible builds will use the trusted server path to download scripts for future builds. If that's the case can you please mention it in this HLD?

Reply:
We have a comment on that on line 31/ Do you think we should have a further explanation?